### PR TITLE
New project layout, Pod 1.0+Carthage+SPM [WIP]

### LINKED ISSUE #79: Create a full test
We should have a repo which is the result of pod-template which specific arguments. Then a test that runs though the same options and then does a diff on output.

Similar to https://github.com/QueryKit/mogenerator-template/blob/master/Makefile#L7-8. We don't need any complex ruby, simple diff does the job.

This should give us more visibility on the [output and how it looks](https://github.com/QueryKit/mogenerator-template/tree/master/Example), and also allows us to know when we unintentionally break something.

### LINKED ISSUE #131: Recommended project settings
Test case:

1. `pod lib create PROJECT`
2. Open XCode on the workspace
3. Product -> Run

Expected:

Builds with no errors

Actual:

A warning is generated for Pods.xcodeproj titled "Update to recommended settings"

screen shot 2015-11-01 at 4 00 41 pm

### LINKED ISSUE #157: Achieve full compatibility with Swift Package Manager
Update the template layout so that libraries started with `pod-template` will automatically be usable from Swift Package Manager.

TEST CASES:

  * Set up a project with Swift Package Manager like at https://swift.org/package-manager/#example-usage and ensure this library is usable
  * Set up a project with Carthage and ensure no regressions

### PR DESCRIPTION
**Pull request by fulldecent**
## Wording:

 * pod-template -- this project
 * template -- a model that can have things changed
 * vendored -- a chosen template with variable substitutions and other updates applied and ready for making an iOS framework

## pod-template progress:

- [x] Make vendoring work
- [x] Set up testing with rake/rspec to validate the vendoring process (fixes #79)
- [x] Set up Travis CI
- [x] Update variable substitution process, remove unused code

## baseline template progress: (new feature)

- [x] Move all files common to Swift and Objective-C into a "baseline" template
- [x] Update to latest gitignore
- [x] Supports Swift Package Manager format (fixes #157)

## swift template progress:

- [x] Create a hello world example (this is a change), it actually does something
- [x] Remove FDChessboardView and my name from project, use variable substitution
- [x] Remove XCode warnings (fixes #131)

## objective-c template progress: [HELP REQUESTED]

- [ ] Translate the current Swift template into Objective-C

## other blockers:

**PLEASE HELP REVIEW AND ADD BLOCKERS HERE**

- [ ] Confirm each set of options works [HELP REQUESTED]

### CODE REVIEW

**Comment by orta** [comment]
So, this removes a bunch of features that are not optional, like all the questions, swift / objc support, library choices, optional example applications all of which are blockers on this PR.

Other notes, 
- I don't want the current README for their library changing. 
- I also think at first we shouldn't ship a `package.swift` file, there's no point in shipping one that won't work by default simply because AFNetworking does - it'll totally throw off anyone trying to know if something supports SPM or not when there's thousands of libraries that have an SPM package but it's not set up or configured right 

It's got a while to go, but I'm excited to see progress 👍

**Comment by fulldecent** [comment]
I fixed the README and removed explicit SPM support.

Also added some VERY BASIC travis CI and testing for this repo. I've never used rspec (or ruby) before but this should be good minimally viable and others can improve :-). There is one error related to CocoaPods but I think we can still commit with that error.

**Comment by fulldecent** [comment]
Update issue and added work list. Requesting review and help from anyone interested.

**Comment by fulldecent** [comment]
Right now everything should be working except for the objective-c template.

Please anyone assist with testing this and commenting on anything that it broken.

**Comment by orta** [comment]
This definitely has a way to go yet, tests crash on my computer, and even the perfect example ends up un-usable. I want to do some work on the CocoaPods-App 1.0 for the next few days, then I can come and give this polish to make it shippable as a PR to you 👍 

There's some nice ideas in here

screen shot 2016-05-10 at 2 56 01 pm

**Comment by fulldecent** [comment]
@orta Thanks for your help!

**Comment by fulldecent** [comment]
This is what I am using to successfully execute the tests:
 
    git clone git@github.com:fulldecent/pod-template.git
    cd pod-template/
    bundle install --jobs=3 --retry=3
    rake

**Comment by segiddins** [comment]
Looks like the `.travis.yml` needs to specify a ruby version in it

**Comment by fulldecent** [comment]
Testing is killing me -- time for bed :-(

**Comment by fulldecent** [comment]
Now testing is somewhat working

**Comment by fulldecent** [comment]
To all: please advise and help. What needs to be done here and how can we do it?

File: .gitignore [hunk 1]
```diff
@@ -26,8 +26,12 @@ Carthage
...
 # http://guides.cocoapods.org/using/using-cocoapods.html#should-i-ignore-the-pods-directory-in-source-control
-# 
+#
 # Note: if you ignore the Pods directory, make sure to uncomment
 # `pod install` in .travis.yml
 #
 # Pods/
+
+Gemfile.lock
+play
```
Comment (orta) [inline .gitignore]: Where does this come from?

File: configurator/ProjectManipulator.rb [hunk 2]
```diff
@@ -89,11 +81,11 @@ def project_folder
...
     def rename_files
       # shared schemes have project specific names
-      scheme_path = project_folder + "/PROJECT.xcodeproj/xcshareddata/xcschemes/"
-      File.rename(scheme_path + "PROJECT.xcscheme", scheme_path +  @configurator.pod_name + "-Example.xcscheme")
+#      scheme_path = project_folder + "/PROJECT.xcodeproj/xcshareddata/xcschemes/"
+#      File.rename(scheme_path + "PROJECT.xcscheme", scheme_path +  @configurator.pod_name + "-Example.xcscheme")
 
       # rename xcproject
-      File.rename(project_folder + "/PROJECT.xcodeproj", project_folder + "/" +  @configurator.pod_name + ".xcodeproj")
+#      File.rename(project_folder + "/PROJECT.xcodeproj", project_folder + "/" +  @configurator.pod_name + ".xcodeproj")
```
Comment (orta) [inline configurator/ProjectManipulator.rb]: Do these want to stay commented?

File: templates/swift/Example/Podfile [hunk 3]
```diff
@@ -1,11 +1,11 @@
+source 'https://github.com/CocoaPods/Specs.git'
```
Comment (orta) [inline templates/swift/Example/Podfile]: Why did you add this? ( It's not needed )

File: templates/swift/Example/Source/AppDelegate.swift [hunk 4]
```diff
@@ -1,9 +1,9 @@
 //
 //  AppDelegate.swift
-//  PROJECT
+//  PROJECT iOS
 //
-//  Created by PROJECT_OWNER on TODAYS_DATE.
-//  Copyright (c) TODAYS_YEAR PROJECT_OWNER. All rights reserved.
+//  Created by Full Decent on 4/25/16.
+//  Copyright © 2016 William Entriken. All rights reserved.
```
Comment (orta) [inline templates/swift/Example/Source/AppDelegate.swift]: ^

File: templates/swift/Example/Source/ViewController.swift [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+//  Created by ${USER_NAME} on ${DATE}.
+//  Copyright © ${YEAR} ${USER_NAME}. All rights reserved.
+//
+
+import UIKit
+import PROJECT
+
+class ViewController: UIViewController {
+    @IBOutlet var chessboard: PROJECT!
+
```
Comment (orta) [inline templates/swift/Example/Source/ViewController.swift]: This isn't a generic, set up template example template yet ^

File: templates/swift/README.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+ * Mutable game state (i.e. can move the pieces)
+ * Animation for piece moves
+ * Highlighting of legal squares for a piece after begin dragging
+ * Premove
+
+
+See Also
+===========
+
+See also Kibitz for Mac which is making a comeback https://github.com/fulldecent/kibitz
```
Comment (orta) [inline templates/swift/README.md]: This is not the README for a new project ^

**Comment by orta** [comment]
Given that this PRs into master, it cannot be merged until it's 100% production ready - as everyone instantly starts using it, there is no deploy process. So it'll need to have the exact same feature set from before. 

You're welcome to PR to a `develop` branch ( I've just pushed master to it right now ) and then I can help out on some of these, but it's still got some work to do 👍

File: .gitignore (hunk 1 above)
Comment (fulldecent) [inline .gitignore]: These updates use Rake (installed by Gemfile) for testing in Travis CI.

File: configurator/ProjectManipulator.rb (hunk 2 above)
Comment (fulldecent) [inline configurator/ProjectManipulator.rb]: Thank you. No, this is deleted in a38d43973dda8e7a1a868989825c43086171581e

File: templates/swift/Example/Podfile (hunk 3 above)
Comment (fulldecent) [inline templates/swift/Example/Podfile]: Removed in ed0103ec67adbc4bbbb7ccc18ff8b1b3a8848fd3

File: templates/swift/Example/Source/AppDelegate.swift (hunk 4 above)
Comment (fulldecent) [inline templates/swift/Example/Source/AppDelegate.swift]: Thank you, fixed at 1319e9ff8e164a12786f1e60d379d18a0be5923d

File: templates/swift/Example/Source/ViewController.swift (hunk 5 above)
Comment (fulldecent) [inline templates/swift/Example/Source/ViewController.swift]: Thank you, fixed with cf7656fe299594d4e1dae6d2a91d1dbce43414b7

File: templates/swift/README.md (hunk 6 above)
Comment (fulldecent) [inline templates/swift/README.md]: Fixed in cf7656fe299594d4e1dae6d2a91d1dbce43414b7

**Comment by fulldecent** [comment]
@orta Thank you for ALL your help. I am closing this PR and have opened another one against the Develop branch per your note.

https://github.com/CocoaPods/pod-template/pull/179
